addons/account/tests/test_account_account.py

```diff
@@ -188,12 +188,47 @@ def test_compute_account_type(self):
         # account_type should not be altered
         alternate_account = self.env['account.account'].search([('account_type', '!=', existing_account.account_type)], limit=1)
         alternate_code = self.env['account.account']._search_new_account_code(
+            start_code=alternate_account.code,
             company=alternate_account.company_id,
-            digits=len(alternate_account.code),
-            prefix=alternate_account.code[:-1])
+            cache=[alternate_account.code])
         new_account.code = alternate_code
         self.assertEqual(new_account.account_type, existing_account.account_type)
 
+    def test_search_new_account_code(self):
+        """ Test whether the account codes tested for availability are the ones we expect. """
+        def get_tested_codes(start_code, tries=4):
+            class StopSearch(Exception):
+                pass
+
+            tested_codes = []
+
+            def mock_search_count(recordset, domain, limit=1):
+                tested_code = domain[0][2]
+                tested_codes.append(tested_code)
+                if len(tested_codes) >= tries:
+                    raise StopSearch()
+                return 1
+
+            with mock.patch('odoo.models.BaseModel.search_count', autospec=True, side_effect=mock_search_count):
+                try:
+                    self.env['account.account']._search_new_account_code(start_code, self.company_data['company'])
+                except StopSearch:
+                    return tested_codes
+
+        # pylint: disable=C0326
```

Comment:
Use the string code instead of the number code, it's easier for simple humans to read :wink: